kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/shape/def/CustomTaskShapeDefTest.java

```diff
@@ -0,0 +1,155 @@
+    private static final String WID_ICON_DATA_3 = "serviceNodeIcon";
+    private static final String DEF_ID_3 = BindableAdapterUtils.getDynamicDefinitionId(CustomTask.class,
+                                                                                       WID_ID_3);
+
+    private static final WorkItemDefinition DEF_1 = new WorkItemDefinition()
+            .setName(WID_ID_1)
+            .setIconDefinition(new IconDefinition().setIconData(WID_ICON_DATA_1));
+
+    private static final double WIDTH = 111d;
+    private static final double HEIGHT = 121d;
+    private static final Bounds BOUNDS = Bounds.create(0, 0, WIDTH, HEIGHT);
+
+    @Mock
+    private WorkItemDefinitionRegistry registry;
+
+    @Mock
+    private Function<String, Glyph> iconDataGlyphGenerator;
+
+    @Mock
+    private View node;
+
+    @Mock
+    private SVGShapeViewImpl view;
+
+    @Mock
+    private SVGPrimitive imageSvgPrimitive;
+
+    @Mock
+    private Picture imagePrimitive;
+
+    @Mock
+    private ImageProxy imageProxy;
+
+    private CustomTaskShapeDef tested;
+    private CustomTask task;
+
+    @Before
+    @SuppressWarnings("unchecked")
+    public void init() throws Exception {
+        this.task = new CustomTask();
+        task.getDimensionsSet().getWidth().setValue(WIDTH);
+        task.getDimensionsSet().getHeight().setValue(HEIGHT);
+        this.task.setName(WID_ID_1);
+        when(node.getDefinition()).thenReturn(task);
+        when(node.getBounds()).thenReturn(BOUNDS);
+        when(registry.items()).thenReturn(Collections.singleton(DEF_1));
+        when(registry.get(eq(WID_ID_1))).thenReturn(DEF_1);
+        when(registry.get(eq(WID_ID_2))).thenReturn(null);
+        when(registry.get(eq(WID_ID_3))).thenThrow(new RuntimeException("Just for Testing"));
+        when(view.getChildren()).thenReturn(Collections.singletonList(imageSvgPrimitive));
+        when(imageSvgPrimitive.getId()).thenReturn(CustomTaskShapeViewHandler.WID_ICON_ID);
+        when(imageSvgPrimitive.get()).thenReturn(imagePrimitive);
+        when(imageSvgPrimitive.getPrimitiveId()).thenReturn(CustomTaskShapeViewHandler.WID_ICON_ID);
+        when(imagePrimitive.getID()).thenReturn(CustomTaskShapeViewHandler.WID_ICON_ID);
+        when(imagePrimitive.getImageProxy()).thenReturn(imageProxy);
+        this.tested = new CustomTaskShapeDef(() -> registry,
+                                             iconDataGlyphGenerator);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testSizeHandler() {
+        tested.newSizeHandler().handle(node, view);
+        verify(view, times(1)).setSize(eq(111d),
+                                       eq(121d));
+    }
+
+    @Test
+    public void testViewHandler() {
+        when(imageProxy.isLoaded()).thenReturn(true);
+        tested.viewHandler().accept(task,
+                                    view);
+        verify(imageProxy, times(1)).load(eq(WID_ICON_DATA_1));
+    }
+
+    @Test
+    public void testGetGlyph() {
```

Comment:
Same as on old stream, could we make this separate test cases?

Reply:
as on the old stream too, waiting for resolving thsi comment in some way for merging

Reply:
Done. Thanks @domhanak.